app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -326,26 +326,77 @@ def import_enabled_repositories(repos)
       end
 
       def import_module_streams(module_streams)
-        streams = {}
-        module_streams.each do |module_stream|
-          stream = AvailableModuleStream.create_or_find_by!(name: module_stream["name"],
-                                               context: module_stream["context"],
-                                               stream: module_stream["stream"])
-          streams[stream.id] = module_stream
-        end
-        sync_available_module_stream_associations(streams)
-      end
-
-      def sync_available_module_stream_associations(new_available_module_streams)
+        Rails.logger.debug "INSERT_ALL-------------------------------------"
+        Rails.logger.debug module_streams.first(3)
+        streams = module_streams.map do |module_stream|
+          {
+            name: module_stream["name"],
+            stream: module_stream["stream"],
+            context: module_stream["context"]
```

Comment:
AvailableModuleStreams also have `profiles`:

```
[6] pry(main)> ::Katello::AvailableModuleStream
=> Katello::AvailableModuleStream(id: integer, name: string, stream: string, context: string, profiles: )
```

Reply:
Indeed they do. I wonder how `profiles` is getting set, since we've never done so here.

Reply:
I think it may be unused!